src/NodaTime.Test/Text/TypeConvertersTest.cs

```diff
@@ -116,18 +118,30 @@ public void HasConverter(Type type)
         public void Period_Roundtrip(int years, int months, int weeks, int days, long hours, long minutes, long seconds, long milliseconds, long ticks, long nanoseconds, string text) =>
             AssertRoundtrip(text, new Period(years, months, weeks, days, hours, minutes, seconds, milliseconds, ticks, nanoseconds));
 
-        private static void AssertRoundtrip<T>(string input, T expected)
+        [Test]
+        [TestCase(2019, 1, 1, 4, 59, "Europe/Paris", "2019-01-01T04:59:00 Europe/Paris (+01)")]
+        [TestCase(2020, 2, 29, 23, 59, "America/Los_Angeles", "2020-02-29T23:59:00 America/Los_Angeles (-08)")]
+        [TestCase(2018, 12, 31, 13, 30, "Asia/Kathmandu", "2018-12-31T13:30:00 Asia/Kathmandu (+05:45)")]
+        public void ZonedDateTime_Roundtrip(int year, int month, int day, int hour, int minute, string zoneId, string text)
+        {
+            Assert.AreSame(DateTimeZoneProviders.Tzdb, DateTimeZoneProviders.ForTypeConverter,
+                "Expected no other test to change DateTimeZoneProviders.ForTypeConverter");
+            var zone = DateTimeZoneProviders.Tzdb[zoneId];
+            AssertRoundtrip(text, new LocalDateTime(year, month, day, hour, minute).InZoneStrictly(zone));
+        }
+
+        private static void AssertRoundtrip<T>(string textEquivalent, T nodaValue)
         {
             var converter = TypeDescriptor.GetConverter(typeof(T));
-            var actual = (T)converter.ConvertFrom(input);
+            var valueFromConverter = (T)converter.ConvertFrom(textEquivalent);
 
-            Assert.NotNull(actual);
-            Assert.AreEqual(expected, actual);
+            Assert.NotNull(valueFromConverter);
```

Comment:
I know that this is pre-existing, but is it useful to do a non-null check immediately before an equality check? It seems like just calling `AreEqual()` would at least have the potential to give a more useful message in the event of a null?

Or, if you're concerned about `nodaValue` being null, seems like it would be reasonable to add that as a precondition at the start of the method?

(Likewise below.)

Reply:
Nope, I think it's fine to remove these - done.